### PR TITLE
Help attribute manager learn node "remoteness" more quickly

### PR DESCRIPTION
**Pull request by kgaillot**
pacemaker-attrd can write out a remote node's attributes to the CIB only if the node is in its remote peer cache. Unlike the controller, pacemaker-attrd does not manage the remote peer cache via parsing the CIB; instead, it adds nodes to the cache after certain events, the most important being the controller setting the probed attribute when a remote node comes up.

If the cluster restarts on a cluster node, it will get a copy of all current attributes via a sync request to other nodes. However, it will not know that remote nodes are remote, and if the remote nodes are already up, it won't easily learn it. If the node later becomes DC, its scheduler will not see the remote nodes' attributes. If the cluster includes unfencing, this can lead to an unfencing loop.

The main solution is to indicate a node's "remoteness" in replies to the sync request. That way, the cluster node can learn existing remote nodes' "remoteness" as soon as it rejoins the cluster. There are a few other (less important) places where we can pass along the information as well.

### CODE REVIEW

**Comment by kgaillot** [comment]
@clumens , can you review this when you get a chance? Some of it may be hard to follow without knowing how pacemaker-attrd works but most of it is straightforward.

**Comment by clumens** [comment]
This all looks good to me.
